src/Agent.Listener/Configuration/WindowsRegistryHelper.cs

```diff
@@ -0,0 +1,206 @@
+                case WellKnownRegistries.LegalNoticeCaption :
+                    return _registryManager.GetKeyValue(RegistryConstants.RegPaths.LegalNotice, RegistryConstants.KeyNames.LegalNoticeCaption);
+                case WellKnownRegistries.LegalNoticeText :
+                    return _registryManager.GetKeyValue(RegistryConstants.RegPaths.LegalNotice, RegistryConstants.KeyNames.LegalNoticeText);
+                default:
+                   return null;
+            }
+        }
+
+        public void DeleteRegistry(WellKnownRegistries targetRegistry)
+        {
+            switch(targetRegistry)
+            {                
+                //machine specific registry settings
+                case WellKnownRegistries.AutoLogonCount :
+                     _registryManager.DeleteKey(RegistryConstants.RegPaths.AutoLogon, RegistryConstants.KeyNames.AutoLogonCount);
+                     break;
+                case WellKnownRegistries.AutoLogonPassword :
+                     _registryManager.DeleteKey(RegistryConstants.RegPaths.AutoLogon, RegistryConstants.KeyNames.AutoLogonPassword);
+                     break;                
+                default:
+                   throw new InvalidOperationException("Delete registry is called for a undocumented registry");
+            }
+        }
+
+        private string GetUserRegistryRootPath(string sid)
+        {
+            return string.IsNullOrEmpty(sid) ?
+                RegistryConstants.CurrentUserRootPath :
+                String.Format(RegistryConstants.DifferentUserRootPath, sid);
+        }
+    }
+    
+    public enum WellKnownRegistries
+    {
+        ScreenSaver,
+        ScreenSaverDomainPolicy,
+        AutoLogon,
+        AutoLogonUserName,
+        AutoLogonDomainName,
+        AutoLogonCount,
+        AutoLogonPassword,
+        StartupProcess,
+        ShutdownReason,
+        ShutdownReasonUI,
+        LegalNoticeCaption,
+        LegalNoticeText
+    }
+
+    public struct RegistryConstants
+    {
+        public const string CurrentUserRootPath = @"HKEY_CURRENT_USER";
+        public const string DifferentUserRootPath = @"HKEY_USERS\{0}";
+
+        public struct RegPaths
+        {
+            public const string ScreenSaver = @"{0}\Control Panel\Desktop";
+            public const string ScreenSaverDomainPolicy = @"{0}\Software\Policies\Microsoft\Windows\Control Panel\Desktop";
+            public const string StartupProcess = @"{0}\SOFTWARE\Microsoft\Windows\CurrentVersion\Run";
+            public const string AutoLogon = @"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon";
+            public const string ShutdownReasonDomainPolicy = @"HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows NT\Reliability";
+            public const string LegalNotice = @"HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon";
```

Comment:
When the process is going to run as 64bit all the reg are from 64bit hive (i.e., the default path). We don't need to handle the case explicitly. Please let me know if I am missing something here.